tests/app/test_jinja_filters.py

```diff
@@ -859,3 +859,27 @@ def test_format_repeated_summaries_delimiters(self):
         self.autoescape_context = Mock(autoescape=True)
         output = format_repeating_summary(self.autoescape_context, [['', '51 Testing Gardens', '', 'Bristol', 'BS9 1AW']], delimiter=', ')
         self.assertEqual(output, '<ul><li>51 Testing Gardens, Bristol, BS9 1AW</li></ul>')
+
+    def test_format_address_list_undefined_values(self):
+        address1 = [Undefined(), Undefined(), Undefined(), Undefined(), Undefined()]
+        address2 = ['123', 'Testy', 'Place', 'Newport', 'NP5 7AR']
+        self.assertEqual('123<br />Testy<br />Place<br />Newport<br />NP5 7AR',
+                         format_address_list(address1, address2))
+
+    def test_format_address_list_None_value(self):
+        address1 = [None, None, None, None, None]
+        address2 = [None, None, None, None, None]
+        with self.assertRaises(Exception):
+            format_address_list(address1, address2)
+
+    def test_format_address_list_no_metadata(self):
+        address1 = ['44', 'Testing', 'Gardens', 'Swansea', 'SA1 1AA']
+        address2 = []
+        self.assertEqual('44<br />Testing<br />Gardens<br />Swansea<br />SA1 1AA',
+                         format_address_list(address1, address2))
+
+    def test_format_address_list(self):
+        address1 = ['44', 'Testing', 'Gardens', 'Swansea', 'SA1 1AA']
+        address2 = ['123', 'Testy', 'Place', 'Newport', 'NP5 7AR']
```

Comment:
Can you create a test where you have missing values from both that don't match?
eg 
```
address1 = ['44', None, 'Gardens', None, 'SA1 1AA']
address2 = [None, 'Testy', None, 'Newport', None]
```

Reply:
Or would the `None`s be `''`?

Reply:
Both come back as `''` unless all the answers the answers were `None` or `''` then an exception would be raised